internal/datastore/memdb/namespace.go

```diff
@@ -19,6 +19,10 @@ const (
 )
 
 func (mds *memdbDatastore) WriteNamespace(ctx context.Context, newConfig *v0.NamespaceDefinition) (datastore.Revision, error) {
+	if mds.db == nil {
+		panic("memdb datastore already closed")
```

Comment:
this is panic but the rest are errors, which do we want? I'd say panic -- that's what other things in the stdlib do (even a double Close() on resources in the stdlib panic)